### PR TITLE
Fixes #28869 - change the description of the scoped loc and org

### PR DESCRIPTION
**Pull request by shiramax**

### CODE REVIEW

**Comment by xprazak2** [comment]
[#28869](https://projects.theforeman.org/issues/28869) is about `hammer location create --organization-id x` not adding the newly created location to an organization. Looking at the code and commit message, maybe this is meant to fix a different issue?

**Comment by shiramax** [comment]
@xprazak2 I use the script for cloning the bug to Redmine, so it just copied everything without my comment, so I will add it here and also in redmine:

organization_id can be a bit confusing, 
Normally you would expect to set the organization but it's not the case. 
organization_id can only set the scope of the request. 
like, In the UI, you have those two buttons of the location and organization, and they are setting the scope of all foreman actions...
organization_id will not set the organization just the scope of it.
so, the best approach here is to change the help because it could be a bit confusing.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 1]
```diff
@@ -46,10 +51,20 @@ def self.s_name(description, options={})
       :hostgroup =>        [ s_name(_("Hostgroup name")), s("title", _("Hostgroup title"), :editable => false) ],
       # :image =>            [],
       :interface =>        [],
-      :location =>         [ s_name(_("Location name")), s("title", _("Location title"), :editable => false) ],
+      :location =>         [  s("name", _("Location Name, Set the current location context for the request")),
+                              s("title", _("Location title, Set the current location context for the request" ),:editable => false),
+                              s("id", _("Set the current location context for the request"),:editable => false, :format => HammerCLI::Options::Normalizers::Number.new),
+
+
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/id_resolver.rb]: Redundant extra lines.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 2]
```diff
@@ -46,10 +51,20 @@ def self.s_name(description, options={})
...
+      ],
       :medium =>           [ s_name(_("Medium name")) ],
       :model =>            [ s_name(_("Model name")) ],
-      :organization =>     [ s_name(_("Organization name")), s("title", _("Organization title"), :editable => false) ],
+      :organization =>     [ s("name", _("Set the current organization context for the request")),
+                             s("title", _("Set the current organization context for the request"),:editable => false),
+                             s("id", _("Set the current organization context for the request"),:editable => false, :format => HammerCLI::Options::Normalizers::Number.new),
+
+
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/id_resolver.rb]: Dtto extra lines.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 3]
```diff
@@ -46,10 +51,20 @@ def self.s_name(description, options={})
       :hostgroup =>        [ s_name(_("Hostgroup name")), s("title", _("Hostgroup title"), :editable => false) ],
       # :image =>            [],
       :interface =>        [],
-      :location =>         [ s_name(_("Location name")), s("title", _("Location title"), :editable => false) ],
+      :location =>         [  s("name", _("Location Name, Set the current location context for the request")),
+                              s("title", _("Location title, Set the current location context for the request" ),:editable => false),
+                              s("id", _("Set the current location context for the request"),:editable => false, :format => HammerCLI::Options::Normalizers::Number.new),
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/id_resolver.rb]: Missing space before `:editable`.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 4]
```diff
@@ -46,10 +51,20 @@ def self.s_name(description, options={})
...
+
+
+      ],
       :medium =>           [ s_name(_("Medium name")) ],
       :model =>            [ s_name(_("Model name")) ],
-      :organization =>     [ s_name(_("Organization name")), s("title", _("Organization title"), :editable => false) ],
+      :organization =>     [ s("name", _("Set the current organization context for the request")),
+                             s("title", _("Set the current organization context for the request"),:editable => false),
+                             s("id", _("Set the current organization context for the request"),:editable => false, :format => HammerCLI::Options::Normalizers::Number.new),
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/id_resolver.rb]: Dtto before `:editable`.

File: lib/hammer_cli_foreman/id_resolver.rb [hunk 5]
```diff
@@ -96,14 +111,13 @@ def scoped_options(scope, options, mode = nil)
 
       option_names = []
       if (mode.nil? || mode == :single)
-        option_names << "id"
         option_names += searchables(resource).map { |s| s.name }
+        option_names << "id" unless  option_names.include?("id")
```
Comment (ofedoren) [inline lib/hammer_cli_foreman/id_resolver.rb]: Redundant extra space after `unless`.

**Comment by shiramax** [comment]
@ofedoren done:)

**Comment by ofedoren** [comment]
@shiramax, the tests are red, but locally they pass. Could you please rebase to ensure everything is ok?

**Comment by shiramax** [comment]
@ofedoren tests are green now :)

**Comment by ofedoren** [comment]
@shiramax, thanks! Merging :)

**Pull request merged**